### PR TITLE
Merging in VB Translations of Design and Performance diagnostics

### PR DESCRIPTION
**Pull request by jwooley**
Check-in includes translations of the design and performance diagnostic code fixes along with a sample base class for analysis and missing testing infrastructure from the original VB project template. Future pull requests will be more atomic.

### CODE REVIEW

**Comment by giggio** [comment]
Wow, that is a lot of work. Good job, @jwooley! I am looking into it right now.

File: src/CSharp/CodeCracker/Performance/RemoveWhereWhenItIsPossibleCodeFixProvider.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public sealed override FixAllProvider GetFixAllProvider()
             var whereMemberAccess = whereInvoke.ChildNodes().OfType<MemberAccessExpressionSyntax>().FirstOrDefault();
             var nextMethodMemberAccess = nextMethodInvoke.ChildNodes().OfType<MemberAccessExpressionSyntax>().FirstOrDefault();
             var newNextMethodInvoke = SyntaxFactory.InvocationExpression(
-                SyntaxFactory.MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, whereMemberAccess.Expression, nextMethodMemberAccess.Name),
+                SyntaxFactory.MemberAccessExpression( SyntaxKind.SimpleMemberAccessExpression, whereMemberAccess.Expression, nextMethodMemberAccess.Name),
```
Comment (viniciushana) [inline src/CSharp/CodeCracker/Performance/RemoveWhereWhenItIsPossibleCodeFixProvider.cs]: I guess this was unintended, right?

**Comment by viniciushana** [comment]
A whole lot of work indeed, thank you!

File: src/CSharp/CodeCracker/Performance/RemoveWhereWhenItIsPossibleCodeFixProvider.cs (hunk 1 above)
Comment (giggio) [inline src/CSharp/CodeCracker/Performance/RemoveWhereWhenItIsPossibleCodeFixProvider.cs]: I already caught that. I reverted all changes to the C# files.

Comment (jwooley) [inline src/CSharp/CodeCracker/Performance/RemoveWhereWhenItIsPossibleCodeFixProvider.cs]: Sorry. I should have audited the checkin a bit closer. Didn't intend to update any .cs files in this update. I did notice some potential issues/gaps in some of the CS scenarios, but didn't make modifications to them yet.

**Comment by giggio** [comment]
@jwooley I have merged it. See commit c79b29b to understand what I changed. Here is a summary from the commit log:

>Revert incorrectly changed C# code
>Remove analyzer base class (still working on that idea)
>Remove Diagnostic id modules (will implement them in a different way)
>Reorder fixes so they match the C# ones
>Make solution match C# solution
>Remove TestCustomWorkspace (not needed in CTP5)
>Removed nuget references on vsix project
>Added namespaces to analyzers, fixes and tests
>Updated names (from sealed to NotInherittable, etc)
>Updated AssemblyInfo
>Removed dead code
>Updated nuget references that were outdated
>removed unnecessary .gitkeep files
>Add shared to C# code fixes that were missing it

I have removed the use of base classes and diagnostic ids on central locations. We are still looking for a definition for those. We will discuss them soon and then we can apply them to both C# and VB.

@jwooley tks a lot for the effort. Now the VB project has real meaning. Let's keep it up. I hope other contributors come to join you! :raised_hands:

**Comment by jwooley** [comment]
Thanks for all of the fixes. I intended to get to some of them but wanted to get something out there as a starting point. I’m planning on continuing on with some other categories. Does it make sense to add separate work items for each analyzer moving forward and have people sign up to work on them so that we avoid duplicate efforts (assuming others are working on translations at this point)?






Sent from Windows Mail





From: Giovanni Bassi
Sent: ‎Saturday‎, ‎January‎ ‎24‎, ‎2015 ‎10‎:‎49‎ ‎AM
To: 'code-cracker/code-cracker'
Cc: Jim Wooley





@jwooley I have merged it. See commit c79b29b to understand what I changed. Here is a summary from the commit log:


Revert incorrectly changed C# code
Remove analyzer base class (still working on that idea)
Remove Diagnostic id modules (will implement them in a different way)
Reorder fixes so they match the C# ones
Make solution match C# solution
Remove TestCustomWorkspace (not needed in CTP5)
Removed nuget references on vsix project
Added namespaces to analyzers, fixes and tests
Updated names (from sealed to NotInherittable, etc)
Updated AssemblyInfo
Removed dead code
Updated nuget references that were outdated
removed unnecessary .gitkeep files
Add shared to C# code fixes that were missing it

I have removed the use of base classes and diagnostic ids on central locations. We are still looking for a definition for those. We will discuss them soon and then we can apply them to both C# and VB.

@jwooley tks a lot for the effort. Now the VB project has real meaning. Let's keep it up. I hope other contributors come to join you!  :raised_hands:

**Comment by giggio** [comment]
@jwooley I recommend you open a PR as soon as you are done with each of them. It is a lot of work to review such a large PR. If you open more PRs they get in quickly, and more people can help reviewing.

**Comment by ljw1004** [comment]
Minor nitpick: following should use VB capitalization for NameOf. Or you could use SyntaxFactory.NameOfExpression.

Dim newNameof = SyntaxFactory.ParseExpression(String.Format("nameof({0})",

**Comment by giggio** [comment]
Good one, ok, will fix it.

**Comment by giggio** [comment]
Fixed on cbed244.

**Pull request merged**